### DISCUSSION TITLE
TWCManager v1.3.0

### DISCUSSION THREAD

**Discussion by ngardiner** [ngardiner/TWCManager#517] [Announcements]
**Bugfixes**

- (@dtiefnig) - Fix issue with logic around detecting instances of multi-phase system incompatibility
- (@dtiefnig) - Fix issue with zero-byte VINs causing infinite VIN fetch retries
- (@MikeBishop) - Fix rounding in calculation of current offer when calculating delta to new offer to improve accuracy
- (@ngardiner) - Adjust voltage / phase detection to handle combinations of 1/2/3 phase systems, as long as all slave TWCs have the same number of active phases
- (@ccutrer) - Fix openhab integration to properly parse floats
- (@ngardiner) - Detect issues with serial connection (RS485) and automatically reconnect
- (@RitchieB2B) - Fix MQTT EMS module (missing brokerPort)
- (@RitchieB2B) - Fix large steps in ChargeNow duration in the web interface
- (@mattiasclaesson) - Remove unnecessary kWh suffix from web UI

**Features**

- (@hopfi2k) - Major "Modern Theme" update. Switched to Bootstrap 5.02, bug fixes, now fully responsive
- (@ngardiner) - Add Home Location view and modification controls to settings page
- (@ngardiner) - Add MQTT EMS module (currently pre-release for testing)
- (@dtiefnig) - Add switch to output plain text Console output (turn off color)
- (@dtiefnig) - Unify single and multi-car charge at home behaviour
- (@dtiefnig) - Add Modbus TCP support for SolarEdge EMS module
- (@dtiefnig / @MikeBishop) - Update Tesla API charging states
- (@dtiefnig) - Reference entrypoint script from within Docker container to remove dependency on external source tree
- (@ngardiner) - Add Current Policy as status value (to be published to HASS / MQTT)
- (@MikeBishop) - Fix API interactions impacted by Tesla's vehicle_data endpoint changes
- (@MikeBishop) - Implement improved VIN Check routine
- (@RitchieB2B) - Add option to leverage Tesla API for lower amp control, otherwise leverage TWC
- (@RitchieB2B) - Add DSM Reader EMS Module
- (@cods4) - Update to documentation

<hr /><em>This discussion was created from the release <a href='https://github.com/ngardiner/TWCManager/releases/tag/v1.3.0'>TWCManager v1.3.0</a>.</em>

**Comment by ChutneyMary** [comment]
I'm a bit confused by the versioning number of the latest release. My existing installation is at v1.3.0. Is there a reason why this release has retained v1.3.0?

**Reply by ngardiner** [reply]
The previous release was v1.2.6, however if you have been using the dev version (pulling the latest commits from the main branch) you would have seen v1.3.0 in the GUI.

I'll look at adding a visual indicator to distinguish a release version from the development tree.

**Reply by ChutneyMary** [reply]
Thanks. I missed the distinction between Release and Main. Notwithstanding, I still can't successfully install either with both versions having the Exception error described at [#521](https://github.com/ngardiner/TWCManager/issues/521)
